contrib/lib/src/templates/mod.rs

```diff
@@ -329,12 +315,13 @@ impl Template {
 /// rendering fails, an `Err` of `Status::InternalServerError` is returned.
 impl Responder<'static> for Template {
     fn respond_to(self, req: &Request) -> response::Result<'static> {
-        let ctxt = req.guard::<State<Context>>().succeeded().ok_or_else(|| {
+        let cm = req.guard::<State<ContextManager>>().succeeded().ok_or_else(|| {
             error_!("Uninitialized template context: missing fairing.");
             info_!("To use templates, you must attach `Template::fairing()`.");
             info_!("See the `Template` documentation for more information.");
             Status::InternalServerError
         })?;
+        let ctxt = cm.get();
 
         let (render, content_type) = self.finalize(&ctxt)?;
```

Comment:
Presumably we can't call `cm.get()` here directly because of some borrow checking issues?

Reply:
Yes, and I think I may have finally figured out why. The `State` type implements `Deref`, whose lifetime is tied to the `State` object itself. Using `.inner()` returns a longer lifetime and "escapes" out of the temporary `State` value.